agent/fw_drupal8.c

```diff
@@ -605,12 +605,128 @@ NR_PHP_WRAPPER(nr_drupal94_invoke_all_with_clean) {
 NR_PHP_WRAPPER_END
 #endif  // OAPI
 
+#define NR_FREE_HOOK_MEM \
+  nr_free(hook_str);     \
+  nr_free(class_str);    \
+  nr_free(method_str);   \
+  nr_free(module_str);   \
+  nr_free(hookpath);
+
+/*
+ * Purpose: Instrument Drupal Attribute Hooks for Drupal 11.1+
+ *
+ * Params: 1. A zval pointer to the moduleHandler instance in use by Drupal.
+ *
+ * Return: bool
+ *
+ */
+static bool nr_drupal_hook_attribute_instrument(zval* module_handler) {
+  zval* hook_implementation_map = NULL;
+
+  nr_php_string_hash_key_t* hook_key = NULL;
+  zval* hook_val = NULL;
+  nr_php_string_hash_key_t* class_key = NULL;
+  zval* class_val = NULL;
+  nr_php_string_hash_key_t* method_key = NULL;
+  zval* module_val = NULL;
+  zend_ulong key_num = 0;
+
+  char* hook_str = NULL;
+  char* class_str = NULL;
+  char* method_str = NULL;
+  char* module_str = NULL;
+  char* hookpath = NULL;
+
+  hook_implementation_map = nr_php_get_zval_object_property(
+      module_handler, "hookImplementationsMap");
+  if (hook_implementation_map) {
+    if (nr_php_is_zval_valid_array(hook_implementation_map)) {
+      ZEND_HASH_FOREACH_KEY_VAL(Z_ARRVAL_P(hook_implementation_map), key_num,
+                                hook_key, hook_val) {
+        (void)key_num;
+        if ((NULL == hook_key) || (0 == nr_php_is_zval_valid_array(hook_val))) {
+          nrl_warning(NRL_FRAMEWORK,
+                      "hookImplementationsMap[hook = %s]: invalid value",
+                      NRSAFESTR(ZEND_STRING_VALUE(hook_key)));
+          NR_FREE_HOOK_MEM
+          return false;
+        }
+
+        ZEND_HASH_FOREACH_KEY_VAL(Z_ARRVAL_P(hook_val), key_num, class_key,
```

Comment:
[This comment](https://github.com/newrelic/newrelic-php-agent/pull/1030/files#r1972313208) applies here as well.

Reply:
Addressed with https://github.com/newrelic/newrelic-php-agent/commit/6b3d5b1536f40234b2de87bca522ed11a4637118